README.md
# My protfolio website

[![Netlify Status](https://api.netlify.com/api/v1/badges/df232a91-e0ac-4c89-ae62-f91528fb23a3/deploy-status)](https://app.netlify.com/sites/cranky-tesla-dab923/deploys)

This is my portfolio website.

It is based on [Gatsby starter Strata](https://www.gatsbyjs.org/starters/codebushi/gatsby-starter-strata/) which is based on the Strata site template, designed by [HTML5 UP](https://html5up.net/strata).

## Requirements

- Node.js: ^12.3.0
- Yarn: ^1.22.0

## Installation

Run `yarn` to install all dependencies.

## Run in development mode

Run `yarn develop` to launch the website in development mode.

## Run in production mode

1. Run `yarn build` in order to build the production bundle.
2. Run `yarn serve` to launch the website in production mode.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [1.5.0] - 2020-03-30

### Added [1.5.0]

- Use Netlify's File-based configuration

### Changed [1.5.0]

- Update README.md
- Update Web App Manifest

## [1.4.0] - 2020-03-29

### Changed [1.4.0]

- Use Yarn
- Use Gatsby's ESLint config
- Update dependencies
- Use React functional components

## [1.3.1] - 2019-11-10

### Changed [1.3.1]

- Update dependencies

## [1.3.0] - 2019-06-16

### Added [1.3.0]

- Add Google Analytics
- Add robots.txt
- Add title property to a tags

### Changed [1.3.0]

- Update title and h1 tags
- Pass images to render gallery function

### Removed [1.3.0]

- Remove yargs package

## [1.2.0] - 2019-06-15

### Changed [1.2.0]

- Update dependencies
- Update react-images package and its usage
- Update website content
  - Update my current position to "Fullstack JavaScript Developer"

## [1.1.0] - 2019-06-10

### Changed [1.1.0]

- Update dependencies
- Update website content
  - Update my title to "Fullstack JavaScript Developer"
  - Remove the anchor in my photo
- Make the project prettier

## [1.0.0] - 2019-04-25

### Added [1.0.0]

- Add my portfolio website

[unreleased]: https://github.com/GhassenRjab/portfolio/compare/v1.0.0...HEAD
[1.0.0]: https://github.com/GhassenRjab/portfolio/releases/tag/v1.0.0


gatsby-config.js
module.exports = {
  siteMetadata: {
    title: "Gatsby Starter - Strata by HTML5 UP",
    author: "Hunter Chang",
    description: "A Gatsby.js Starter based on Strata by HTML5 UP",
    siteUrl: "https://www.ghassenrjab.tn",
  },
  plugins: [
    "gatsby-plugin-react-helmet",
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: "Ghassen Rjab's portfolio website",
        short_name: "Ghassen Rjab",
        start_url: "/",
        background_color: "#555259",
        theme_color: "#565952",
        display: "standalone",
        icon: "src/assets/images/website-icon.png",
      },
    },
    "gatsby-plugin-sass",
    "gatsby-plugin-offline",
    {
      resolve: "gatsby-plugin-html-attributes",
      options: {
        lang: "en",
      },
    },
    "gatsby-plugin-sitemap",
    "gatsby-plugin-robots-txt",
    {
      resolve: "gatsby-plugin-google-analytics",
      options: {
        trackingId: "UA-142190723-1",
        head: true,
      },
    },
  ],
};


src/components/Footer.js
import React from "react";
import { OutboundLink } from "gatsby-plugin-google-analytics";

const Footer = () => (
  <div id="footer">
    <div className="inner">
      <ul className="icons">
        <li>
          <OutboundLink
            href="https://twitter.com/GhassenRjab"
            className="icon fa-twitter"
            title="Twitter profile"
            aria-label="Twitter profile"
          >
            <span className="label">Twitter</span>
          </OutboundLink>
        </li>
        <li>
          <OutboundLink
            href="https://github.com/GhassenRjab"
            className="icon fa-github"
            title="GitHub profile"
            aria-label="GitHub profile"
          >
            <span className="label">Github</span>
          </OutboundLink>
        </li>
        <li>
          <OutboundLink
            href="https://www.linkedin.com/in/ghassenrjab/"
            className="icon fa-linkedin"
            title="LinkedIn profile"
            aria-label="LinkedIn profile"
          >
            <span className="label">LinkedIn</span>
          </OutboundLink>
        </li>
        <li>
          <a
            href="mailto:rjab.ghassen@gmail.com"
            className="icon fa-envelope-o"
            title="Email address"
            aria-label="Email address"
          >
            <span className="label">Email</span>
          </a>
        </li>
      </ul>
      <ul className="copyright">
        <li>&copy; Gatsby Starter Strata</li>
        <li>
          Design:{" "}
          <OutboundLink href="http://html5up.net" title="HTML5 UP">
            HTML5 UP
          </OutboundLink>
        </li>
      </ul>
    </div>
  </div>
);

export default Footer;


src/components/Gallery.js
import React, { useState } from "react";
import PropTypes from "prop-types";
import Carousel, { Modal, ModalGateway } from "react-images";

const Gallery = ({ images }) => {
  const [lightboxIsOpen, setLightboxIsOpen] = useState(false);
  const [selectedIndex, setSelectedIndex] = useState(0);

  const toggleLightbox = (selectedIndex) => {
    setLightboxIsOpen(!lightboxIsOpen);
    setSelectedIndex(selectedIndex);
  };

  const renderGallery = (images) => {
    if (!images) return;

    const gallery = images.map((obj, i) => {
      return (
        <article className="6u 12u$(xsmall) work-item" key={i}>
          <a
            className="image fit thumb"
            href={obj.source}
            title={obj.caption}
            onClick={(e) => {
              e.preventDefault();
              toggleLightbox(i);
            }}
          >
            <img src={obj.thumbnail} alt={obj.caption} />
          </a>

          <h3>{obj.caption}</h3>
          <p>{obj.description}</p>
        </article>
      );
    });

    return <div className="row">{gallery}</div>;
  };

  return (
    <div>
      {renderGallery(images)}
      <ModalGateway>
        {lightboxIsOpen && (
          <Modal onClose={toggleLightbox}>
            <Carousel currentIndex={selectedIndex} views={images} />
          </Modal>
        )}
      </ModalGateway>
    </div>
  );
};

Gallery.displayName = "Gallery";
Gallery.propTypes = {
  images: PropTypes.array,
};

export default Gallery;


src/pages/index.js
import React from "react";
import Helmet from "react-helmet";

import Layout from "../components/layout";
import Gallery from "../components/Gallery";

import thumb01 from "../assets/images/thumbs/01.jpg";
import thumb02 from "../assets/images/thumbs/02.jpg";
import thumb03 from "../assets/images/thumbs/03.jpg";
import thumb04 from "../assets/images/thumbs/04.jpg";
import thumb05 from "../assets/images/thumbs/05.jpg";
import thumb06 from "../assets/images/thumbs/06.jpg";
import thumb07 from "../assets/images/thumbs/07.jpg";

import full01 from "../assets/images/fulls/01.jpg";
import full02 from "../assets/images/fulls/02.jpg";
import full03 from "../assets/images/fulls/03.jpg";
import full04 from "../assets/images/fulls/04.jpg";
import full05 from "../assets/images/fulls/05.jpg";
import full06 from "../assets/images/fulls/06.jpg";
import full07 from "../assets/images/fulls/07.jpg";

const DEFAULT_IMAGES = [
  {
    source: full01,
    thumbnail: thumb01,
    caption: "Zoe",
    description:
      "Design, set up architecture and develop from scratch Zoe (the chatbot for Quicktext hotels) using Azure Bot Service, Cognitive Services and Node.js.",
  },
  {
    source: full02,
    thumbnail: thumb02,
    caption: "Paristay",
    description:
      "Full-stack development of Paristay using CodeIgniter. Improve SQL queries and site response time. Develop a dashboard dedicated to yield management.",
  },
  {
    source: full03,
    thumbnail: thumb03,
    caption: "WIKI.TOTAL",
    description: "Full-stack development of WIKI.TOTAL using CodeIgniter.",
  },
  {
    source: full04,
    thumbnail: thumb04,
    caption: "Fondation Lenval",
    description: "Front-end development and development of Drupal modules",
  },
  {
    source: full05,
    thumbnail: thumb05,
    caption: "Urofrance",
    description: "Front-end development and development of Drupal modules",
  },
  {
    source: full06,
    thumbnail: thumb06,
    caption: "SmartServices",
    description:
      "Development and maintenance of SmartServices using Java EE (spring, Hibernate and JSF), a POS Web application for restaurants.",
  },
  {
    source: full07,
    thumbnail: thumb07,
    caption: "BipSMS",
    description:
      "Set up and maintain the BipSMS SMS gateway server using Kannel and PlaySMS.",
  },
];

const HomeIndex = () => (
  <Layout>
    <Helmet>
      <title>
        Ghassen Rjab - Fullstack JavaScript Developer and problem solver
      </title>
      <meta
        name="description"
        content="This is the protfolio website of Ghassen Rjab, a Fullstack JavaScript Developer from Sfax, Tunisia. Based in Paris, France now."
      />
    </Helmet>

    <div id="main">
      <section id="one">
        <header className="major">
          <h2>Fullstack JavaScript Developer</h2>
        </header>
        <p>
          Fullstack JavaScript Developer with a rich experience in Chatbot and
          AI development. Creative, multi-skilled and self-contained, who loves
          facing challenges and solving complex problems with his keyboard.
        </p>
      </section>

      <section id="two">
        <h2>Recent Work</h2>

        <Gallery images={DEFAULT_IMAGES} />
      </section>

      <section id="three">
        <h2>Get In Touch</h2>
        <p>Want to hire me or get in touch with me? Send me a message.</p>
        <div className="row">
          <div className="12u">
            <form
              name="contact"
              method="post"
              data-netlify="true"
              data-netlify-honeypot="bot-field"
            >
              <input type="hidden" name="bot-field" />
              <div className="row uniform 50%">
                <div className="6u 12u$(xsmall)">
                  <input type="text" name="name" id="name" placeholder="Name" />
                </div>
                <div className="6u 12u$(xsmall)">
                  <input
                    type="email"
                    name="email"
                    id="email"
                    placeholder="Email"
                  />
                </div>
                <div className="12u">
                  <textarea
                    name="message"
                    id="message"
                    placeholder="Message"
                    rows="4"
                  ></textarea>
                </div>
              </div>
              <ul className="actions">
                <li>
                  <input type="submit" value="Send Message" />
                </li>
              </ul>
            </form>
          </div>
        </div>
      </section>
    </div>
  </Layout>
);

export default HomeIndex;
